## Assignment
1. Create a list with items: -10, 3, 8, 12, -9, 4
2. Sort this list in ascending order
3. Calculate square root of each item value 
4. Print final list - each item value should be integer

In [1]:
a = [-10, 3, 8, 12, -9, 4]

In [2]:
a

[-10, 3, 8, 12, -9, 4]

In [10]:
a.sort()

In [12]:
a

[-10, -9, 3, 4, 8, 12]

In [14]:
2 ** 3

8

In [16]:
4 ** (1 / 2)

2.0

In [34]:
output_list = []
for i in a:
    sqrt_i = i ** (1 / 2)
    print(sqrt_i)
    print(type(sqrt_i))
    output_list.append(sqrt_i)

(1.9363366072701937e-16+3.1622776601683795j)
<class 'complex'>
(1.8369701987210297e-16+3j)
<class 'complex'>
1.7320508075688772
<class 'float'>
2.0
<class 'float'>
2.8284271247461903
<class 'float'>
3.4641016151377544
<class 'float'>


In [32]:
output_list

[(1.9363366072701937e-16+3.1622776601683795j),
 (1.8369701987210297e-16+3j),
 1.7320508075688772,
 2.0,
 2.8284271247461903,
 3.4641016151377544]

In [80]:
len(output_list)

6

In [38]:
c = 3 + 2j

In [40]:
c

(3+2j)

In [44]:
int(c)

TypeError: int() argument must be a string, a bytes-like object or a real number, not 'complex'

In [50]:
c.imag

2.0

In [53]:
c.real

3.0

In [55]:
sqrt_i = -10 ** (1 / 2)

In [57]:
sqrt_i

-3.1622776601683795

In [74]:
type(sqrt_i)

float

### sqrt_i value here is -3.16, but inside for loop this value is in complex

In [63]:
4 ** (1 / 2)

2.0

In [71]:
2 * 3 / 3 # BODMAS

2.0

In [69]:
2 - 4 + 6

4

In [76]:
import math # Module 
#math.sqrt()  # function


## namespace